pxtlib/initprj.ts

```diff
@@ -172,34 +176,33 @@ cache:
     }]
 }
 `
+        }
     }
 
     export function packageFiles(name: string): pxt.Map<string> {
-        let prj = pxt.appTarget.tsprj || pxt.appTarget.blocksprj;
-        let config = U.clone(prj.config);
-        // remove blocks file
-        Object.keys(prj.files)
-            .filter(f => /\.blocks$/.test(f))
-            .forEach(f => delete prj.files[f]);
-        config.files = config.files.filter(f => !/\.blocks$/.test(f));
-
+        const prj = pxt.appTarget.blocksprj || pxt.appTarget.tsprj;
+        const config = U.clone(prj.config);
+        // clean up
+        delete (<any>config).installedVersion;
+        delete config.additionalFilePath;
+        delete config.additionalFilePaths;
+        // (keep blocks files)
+        config.preferredEditor = config.files.find(f => /\.blocks$/.test(f))
+            ? pxt.BLOCKS_PROJECT_NAME : pxt.JAVASCRIPT_PROJECT_NAME;
         config.name = name;
-        // by default, projects are not public
-        config.public = false;
-
-        if (!config.version) {
+        config.public = true;
```

Comment:
Was changing this to public by default to avoid students making private repos and not knowing how to update it or something of the sort?

Reply:
Public false prevents publishing (sharing) i think